tests/migrate/test_context.py

```diff
@@ -309,3 +308,104 @@ def test_missing_localization_file(self):
             to_json(self.ctx.merge_changeset()),
             expected
         )
+
+
+class TestExistingTarget(unittest.TestCase):
+    def setUp(self):
+        # Silence all logging.
+        logging.disable(logging.CRITICAL)
+
+        self.ctx = MergeContext(
+            lang='pl',
+            reference_dir=here('fixtures/en-US'),
+            localization_dir=here('fixtures/pl')
+        )
+
+        try:
+            self.ctx.maybe_add_localization('privacy.dtd')
+        except RuntimeError:
+            self.skipTest('compare-locales required')
+
+    def tearDown(self):
+        # Resume logging.
+        logging.disable(logging.NOTSET)
+
+    def test_existing_target_ftl_missing_string(self):
+        self.ctx.add_transforms('privacy.ftl', 'privacy.ftl', [
+            FTL.Message(
+                id=FTL.Identifier('dnt-learn-more'),
+                value=COPY(
+                    'privacy.dtd',
+                    'doNotTrack.learnMore.label'
+                )
+            ),
+        ])
+
+        expected = {
+            'privacy.ftl': ftl_resource_to_json('''
+        // This Source Code Form is subject to the terms of the Mozilla Public
+        // License, v. 2.0. If a copy of the MPL was not distributed with this
+        // file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+        dnt-description = New Description in Polish
+        dnt-learn-more = Więcej informacji
+            ''')
+        }
+
+        self.maxDiff = None
+        self.assertDictEqual(
+            to_json(self.ctx.merge_changeset()),
+            expected
+        )
+
+    def test_existing_target_ftl_existing_string(self):
+        self.ctx.add_transforms('privacy.ftl', 'privacy.ftl', [
+            FTL.Message(
+                id=FTL.Identifier('dnt-description'),
+                value=COPY(
+                    'privacy.dtd',
+                    'doNotTrack.description'
+                )
+            ),
+
+            # Migrate an extra string to populate the iterator returned by
+            # ctx.merge_changeset(). Otherwise it won't yield anything if the
+            # merged contents are the same as the existing file.
```

Comment:
Looking at how confused I was that that could happen, maybe it's worth explicitly testing that re-applying a transform returns an empty merge_changeset?